.circleci/config.yml

```diff
@@ -138,42 +138,51 @@ jobs:
-                        approval_downstream = list(filter(lambda x: isinstance(x, dict) and approval_name in list(x.values())[0].get('requires'), jobs))
-                        # replace approval with its upstream jobs
-                        for job in approval_downstream:
-                            requires = next(iter(job.values()))['requires']
-                            requires.remove(approval_name)
-                            requires.extend(approval_upstreams)
-                        jobs.remove(approval)
-                with open('complete_config.yml', 'w') as f:
-                    f.write(yaml.dump(d, sort_keys=False))"
-
-      - continuation/continue:
-          configuration_path: complete_config.yml
+            - run:
+                name: Remove approval steps if not pull from forks.
+                command: |
+                  pip install pyyaml==6.0.1
+                  python -c  "import yaml
+                  d = yaml.safe_load(open('complete_config.yml'))
+                  for workflow_name, workflow_definition in d['workflows'].items():
+                      jobs = workflow_definition.get('jobs') if isinstance(workflow_definition, dict) else None
+                      if not jobs: continue
+                  
+                      # find all approvals
+                      approvals = list(filter(lambda x: isinstance(x, dict) and list(x.values())[0].get('type') == 'approval', jobs))
+                      for approval in approvals:
+                          approval_name = next(iter(approval))
+                          approval_upstreams = approval[approval_name].get('requires')
+                          approval_downstream = list(filter(lambda x: isinstance(x, dict) and approval_name in list(x.values())[0].get('requires'), jobs))
+                          # replace approval with its upstream jobs
+                          for job in approval_downstream:
+                              requires = next(iter(job.values()))['requires']
+                              requires.remove(approval_name)
+                              requires.extend(approval_upstreams)
+                          jobs.remove(approval)
+                  with open('complete_config.yml', 'w') as f:
+                      f.write(yaml.dump(d, sort_keys=False))"
+      - when:
+          condition:
+            or:
+              - equal: [ 'main',  << pipeline.git.branch >> ]
+              - << pipeline.git.tag >>
+          steps:
+            - continuation/continue:
+                configuration_path: complete_config.yml
+                parameters: '{ "build-context": "release" }'
+      - when:
+          condition:
+            not:
+              or:
+                - equal: [ 'main',  << pipeline.git.branch >> ]
+                - << pipeline.git.tag >>
+          steps:
+            - continuation/continue:
+                configuration_path: complete_config.yml
+                parameters: '{ "build-context": "pr" }'
```

Comment:
Assuming there is no easy way to assign parameter value based on condition and that's why different conditions are necessary.